python/ray/tests/test_multiprocessing.py

```diff
@@ -68,6 +69,7 @@ def ray_start_4_cpu():
     ray.shutdown()
 
 
+@pytest.mark.skipif(redis_mode(), reason="Redis is alive across clusters.")
```

Comment:
Where is Redis started, why is Redis not cleaned up?

Reply:
Ok, but I think `_setup_redis()` is supposed to clean up Redis when exiting the context. At least we can leave a comment about the issue, and maybe fix it someday..

Reply:
Yes. But the case here is that **within the test**, there is ray.init and ray.shutdown. So across the tests, it's not a problem.